smurfs/src/components/App.js

```diff
@@ -1,22 +1,73 @@
 import React, { Component } from 'react';
 import './App.css';
+import { connect } from 'react-redux';
+import { getSmurfs, deleteSmurf, updateSmurf, editSmurf } from '../actions';
+import Smurfs from './Smurfs';
+import Form from './Form';
+
 /*
  to wire this component up you're going to need a few things.
  I'll let you do this part on your own. 
  Just remember, `how do I `connect` my components to redux?`
  `How do I ensure that my component links the state to props?`
  */
 class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <h1>SMURFS! 2.0 W/ Redux</h1>
-        <div>Welcome to your Redux version of Smurfs!</div>
-        <div>Start inside of your `src/index.js` file!</div>
-        <div>Have fun!</div>
-      </div>
-    );
-  }
-}
+	constructor() {
+		super();
+		this.state = {
+			name: '',
+			age: '',
+			height: '',
+			id: '',
+		};
+	}
+
+	componentDidMount() {
+		this.props.getSmurfs();
+	}
 
-export default App;
+	handleDelete = (e, id) => {
+		e.preventDefault();
+		this.props.deleteSmurf(id);
+	};
+	handleEdit = (e, id) => {
+		e.preventDefault();
+		let selected = this.props.smurfs.find(smurf => smurf.id === id);
+		this.props.editSmurf(selected);
+		// this.setState({
+		// 	name: selected.name,
+		// 	age: selected.age,
+		// 	height: selected.height,
+		// });
+	};
+	handleUpdate = (e, id) => {
+		e.preventDefault();
+		this.props.deleteSmurf(id);
```

Comment:
Your handle update function needs to be updated, I'm not sure you want to call deleteSmurf() here